packages/ckeditor5-engine/src/model/differ.js

```diff
@@ -336,23 +336,27 @@ export default class Differ {
 	 * @returns {Boolean}
 	 */
 	hasDataChanges() {
+		if ( this._changesInElement.size > 0 ) {
+			return true;
+		}
```

Comment:
This was moved to the beginning but it makes sense, if there are any "regular" changes in the differ, return `true`.